metadata-ingestion/src/datahub/ingestion/source/metabase.py

```diff
@@ -0,0 +1,332 @@
+        try:
+            card_type = type_mapping[display_type]
+        except KeyError:
+            self.report.report_warning(
+                key=f"metabase-card-{card_id}",
+                reason=f"Chart type {display_type} not supported. Setting to None",
+            )
+            card_type = None
+
+        return card_type
+
+    def construct_card_custom_properties(self, card_details: dict) -> Dict:
+        result_metadata = card_details.get("result_metadata", [])
+        metrics, dimensions = [], []
+        for meta_data in result_metadata:
+            display_name = meta_data["display_name"] or ""
+            metrics.append(display_name) if "aggregation" in meta_data.get(
+                "field_ref", ""
+            ) else dimensions.append(display_name)
+
+        filters = str(
+            (card_details["dataset_query"].get("query", {})).get("filter", [])
+        )
+
+        custom_properties = {
+            "Metrics": ", ".join(metrics),
+            "Filters": ", ".join(filters),
+            "Dimensions": ", ".join(dimensions),
+        }
+
+        return custom_properties
+
+    def get_datasource_urn(self, card_details):
+        platform, database_name = self.get_datasource_from_id(
+            card_details["database_id"]
+        )
+        query_type = card_details.get("dataset_query", {}).get("type", {})
+        if query_type == "query":
+            source_table_id = (
+                card_details.get("dataset_query", {})
+                .get("query", {})
+                .get("source-table", {})
+            )
+            schema_name, table_name = self.get_source_table_from_id(source_table_id)
+            if table_name:
+                platform_urn = f"urn:li:dataPlatform:{platform}"
+                dataset_urn = (
+                    f"urn:li:dataset:("
+                    f"{platform_urn},{database_name + '.' if database_name else ''}"
+                    f"{schema_name + '.' if schema_name else ''}"
+                    f"{table_name},{self.config.env})"
+                )
+                return dataset_urn
+        else:
+            self.report.report_warning(
+                key=f"metabase-card-{card_details['id']}",
+                reason=f"Cannot create datasource urn from query type: {query_type}",
+            )
+
+        return None
+
+    @lru_cache(maxsize=None)
+    def get_source_table_from_id(self, table_id):
+        dataset_response = self.session.get(
+            f"{self.config.connect_uri}/api/table/{table_id}"
+        ).json()
+
+        schema = dataset_response.get("schema", "")
+        name = dataset_response.get("name", "")
+
+        return schema, name
+
+    @lru_cache(maxsize=None)
+    def get_datasource_from_id(self, datasource_id):
+        dataset_response = self.session.get(
+            f"{self.config.connect_uri}/api/database/{datasource_id}"
+        ).json()
+
+        return dataset_response["engine"], dataset_response["name"]
```

Comment:
Are we sure dataset_response["engine"] will correctly map to DataHub DataPlatform names? They are listed here: https://github.com/linkedin/datahub/blob/master/metadata-service/war/src/main/resources/boot/data_platforms.json

Are we sure that the dataset_response["name"] will map to the DataHub concept of name? Name should be fully qualified coordinates within a given platform. The exact structure can vary by platform.

Reply:
fixed please see [commit 0c03ae4bc9ca7e473bb7c5c7f0ab6f3a1fb446ab](https://github.com/linkedin/datahub/pull/3602/commits/0c03ae4bc9ca7e473bb7c5c7f0ab6f3a1fb446ab)